Automattic/jetpack
# Search: Refactor search input detection logic

Fixes #22947. 

#### Changes proposed in this Pull Request:
Refactors search input selection logic to filter out inputs belonging to forms unrelated to site search.

#### Jetpack product discussion


See #22947.

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* Apply this change to your site with a Search subscription.
* Ensure that the overlay summoning works as expected depending on the configured overlay trigger settings.
* Try creating a post or a page containing a form mimicking a GeoDirectory form (e.g. a form with the name "geodir-listing-search"). Ensure that typing into an input with a name "s" in this form does not trigger the search overlay.
* Repeat the above for a form with a "post" method. Ensure that the overlay is not triggered.
